plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -848,6 +857,23 @@ public static Schema schemaFromMetadata(List<ColumnMetadata> columns)
         return new Schema(icebergSchema.asStructType().fields());
     }
 
+    public static Schema schemaFromMetadata(List<ColumnMetadata> columns, List<String> identifierFields)
+    {
+        Schema schema = schemaFromMetadata(columns);
```

Comment:
Inline the logic of that method here and do the identifierFields look-ups in the existing for loop